serverlessworkflow/modules/ROOT/pages/cloud/custom-ingress-authz.adoc

```diff
@@ -0,0 +1,326 @@
+
+Enter the example project directory that you cloned locally and run the command below:
+
+.Deploying the "Greeting" workflow
+[source,shell,subs="attributes+"]
+----
+kubectl create ns sonataflow
+kubectl apply -f workflow-app/01-sonataflow-greeting.yaml -n sonataflow
+----
+
+You can follow the workflow deployment by running
+
+.Follow the workflow deployment process
+[source,shell,subs="attributes+"]
+----
+kubectl -n sonataflow get workflow/greeting -w
+
+NAME       PROFILE   VERSION   URL   READY   REASON
+greeting             0.0.1           False   WaitingForBuild
+----
+
+=== Configuring the Ingress Route
+
+Once you deploy the {product_name} workflow you can configure and deploy the APISIX Route.
+
+Open the file `workflow-app/02-sonataflow-route.yaml` in the example application you cloned earlier and change the credentials for the `apisix-ingress` client that you created in the Keycloak server:
+
+."Greetings" workflow APISIX Route
+[source,yaml,subs="attributes+"]
+----
+apiVersion: apisix.apache.org/v2
+kind: ApisixRoute
+metadata:
+  name: sonataflow
+  namespace: sonataflow
+spec:
+  http:
+  - name: greeting
+    match:
+      hosts:
+      - local.greeting.sonataflow.org
+      paths:
+      - "/*"
+    backends:
+    - serviceName: greeting
+      servicePort: 80
+    plugins:
+    - name: openid-connect <1>
+      enable: true
+      config:
+        client_id: apisix-ingress
+        client_secret:  <2>
+        discovery: http://keycloak.keycloak.svc.cluster.local:8080/realms/sonataflow/.well-known/openid-configuration
+        scope: profile email
+        bearer_only: true
+        realm: sonataflow
+        introspection_endpoint_auth_method: client_secret_post
+----
+
+<1> The link:{}[OpenID Connect plugin] to make the Ingress connect to Keycloak
+<2> The `apisix-ingress` client credential to be changed
+
+Open the Keycloak server (link:http://keycloak.keycloak.svc.cluster.local:8080[]) and in the realm "sonataflow" click on "Clients", and then on "apisix-ingress".
+
+Click on the tab "Credentials" and copy the "Client Secret":
+
+.Creating the workflow user
+image::cloud/apisix-keycloak/04-client-credentials.png[]
+
+Paste the "Client Secret" into the `ApisixRoute` file `workflow-app/02-sonataflow-route.yaml` in the example application and run:
+
+.Deploy the `ApisixRoute`
+[source,shell,subs="attributes+"]
+----
+kubectl apply -f workflow-app/02-sonataflow-route.yaml -n sonataflow
+----
+
+To this point, you have installed in your cluster the Keycloak and APISIX Ingress server, and deployed the example "Greetings" workflow application.
+
+=== Accessing the Workflow
+
+Access the workflow without a token to see a rejection:
+
+.Directly accessing the workflow without a token
+[source,shell,subs="attributes+"]
+----
+curl -v POST http://127.0.0.1:$\{INGRESS_PORT\}/greeting -H "Content-type: application/json" -H "Host: local.greeting.sonataflow.org" --data '{ "name": "Luke" }' 
```

Comment:
The url may be different on the users environment. So I think it is better to use url instead of port. + I would add the note as we replace the part of the command here.

```suggestion
INGRESS_URL= <1>

curl -v POST $\{INGRESS_URL\}/greeting -H "Content-type: application/json" -H "Host: local.greeting.sonataflow.org" --data '{ "name": "Luke" }' 
```